### ISSUE TITLE
Encoding of path_info is dubious

### ISSUE THREAD

**Issue by kensanata** [jhthorsen/mojolicious-plugin-cgi#23]
I've noticed a problem with `PATH_INFO` if it contains non-ASCII characters. Here's how to reproduce it.

```
#!/usr/bin/env perl
use Mojolicious::Lite;
use Encode qw(decode_utf8);

plugin CGI => {
  route => '/',
  run => sub {
    print "HTTP/1.1 200 OK\r\n";
    print "Content-Type: text/html; charset=UTF-8\r\n";
    print "\r\n";
    my $path_info = decode_utf8($ENV{PATH_INFO});
    my $query_string = $ENV{QUERY_STRING};
    $query_string =~ s/%([0-9a-f][0-9a-f])/chr(hex($1))/egi;
    $query_string = decode_utf8($query_string);
    binmode(STDOUT, ":encoding(UTF-8)");
    print "<body>";
    print "<p>Path Info: $path_info";
    print "<br>Query String: $query_string";
  },
};

app->start;
```

Run it using `perl something.pl daemon` and visit the following URL:
http://127.0.0.1:3000/f%C3%B6%C3%B6%E2%80%99s?g%C3%B6%C3%B6%E2%80%99s

We get a warning (**Wide character in setenv**) but the result is correct:
```
Path Info: /föö’s
Query String: göö’s
```

Now visit this URL:
http://127.0.0.1:3000/f%C3%B6%C3%B6?g%C3%B6%C3%B6

We get no warning but the `PATH_INFO` result is corrupted:
```
Path Info: /f��
Query String: göö
```

I am reminded of this sentence in the `Encode` man page: «When you decode, the Unicode REPLACEMENT CHARACTER, code point U+FFFD, is used.» The setenv warning is another hint. But first, let's make sure that Mojolicious is not at fault. If you use just `Mojolicious::Lite`, there is no problem. Use the same two URLs as above and the following app. There appear to be no problems.

```
#!/usr/bin/env perl
use Mojolicious::Lite;

get '/*path_info' => sub {
  my $c = shift;
  $c->render(text => "Path Info: " . $c->stash('path_info'));
};

app->start;
```

I'll try and submit a pull request in a bit.

**Comment by kensanata** [comment]
Sadly, I was able to write a test that triggers the warning, but I was unable to write a test that's fatal. I'll give this some more thought.

**Comment by kensanata** [comment]
I've replaced the weak test I had with a better one that starts its own server much like the zombies test. There, I can trigger the problem.

**Comment by jhthorsen** [comment]
I merged #24, but I also made some modifications. Please have a look at 2c45be031cfae05b29e7c862cfcba0b6ff8400fe and tell me what you think.

**Comment by jhthorsen** [comment]
I believe this is resolved in 0.36.
